core/src/mindustry/net/BeControl.java

```diff
@@ -49,6 +49,8 @@ public BeControl(){
             try{
                 Fi dest = Fi.get(System.getProperty("becopy"));
                 Fi self = Fi.get(BeControl.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath());
+                
+                self.parent().findAll(f -> f != self).forEach(Fi::delete);
```

Comment:
1. **Do not** use forEach, that is a Java 8 feature.
2. Have you tested this?
3. Do not use reference comparison for files.

Reply:
I have tested this and it works fine, how would you suggest that I avoid deleting the file without comparing references? If it's built with java 14 then whats the issue with using java 8 features?

Reply:
`findAll` creates new file objects, which means reference comparison is meaningless. You're just deleting the entire contents of the folder. [Use equals.](https://stackoverflow.com/questions/7520432/what-is-the-difference-between-and-equals-in-java)

> If it's built with java 14 then whats the issue with using java 8 features?

[Read the contributing guidelines. ](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md#do-not-use-incompatible-java-features-javautilfunction-javaawt) The version built with is irrelevant.